packages/devtools_app/lib/src/screens/vm_developer/vm_developer_tools_screen.dart

```diff
@@ -78,47 +79,67 @@ class VMDeveloperToolsScreenBody extends StatefulWidget {
 class _VMDeveloperToolsScreenState extends State<VMDeveloperToolsScreenBody>
     with
         AutoDisposeMixin,
+        AutomaticKeepAliveClientMixin<VMDeveloperToolsScreenBody>,
         ProvidedControllerMixin<VMDeveloperToolsController,
             VMDeveloperToolsScreenBody> {
+  int _selectedIndex = 0;
+  late List<Widget> _views;
+
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
     initController();
   }
 
+  void onDestinationSelected(int i) {
+    setState(() => _selectedIndex = i);
+    controller.onSelectedIndex(i);
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _views = [
```

Comment:
we don't want to do this in initState. This means that the _views will only be built once, even if a rebuild is triggered further up the tree and we really want all of `VMDeveloperToolsScreenBody` to be rebuilt.

any time you are building widgets in initState in flutter is a code smell. Generally, never build a widget outside of the build lifecycle.

Reply:
Yeah, you're right. I was following an example I found, my bad...